If there is an `IllegalStateException` in the benchmark run then _"everything fails"_ as:
```
[info] java.lang.IllegalStateException: Good!
[info]  at org.enso.runtime/org.enso.interpreter.bench.benchmarks.semantic.ArrayProxyBenchmarks.sumOverComputingProxy(ArrayProxyBenchmarks.java:100)
[info]  at org.enso.runtime/org.enso.interpreter.bench.benchmarks.semantic.jmh_generated.ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.sumOverComputingProxy_avgt_jmhStub(ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.java:232)
[info]  at org.enso.runtime/org.enso.interpreter.bench.benchmarks.semantic.jmh_generated.ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.sumOverComputingProxy_AverageTime(ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.java:173)
[info]  at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[info]  at java.base/java.lang.reflect.Method.invoke(Method.java:565)
[info]  at jmh.core@1.36/org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:475)
[info]  at jmh.core@1.36/org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:458)
[info]  at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[info]  at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
[info]  at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[info]  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
[info]  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
[info]  at java.base/java.lang.Thread.run(Thread.java:1447)
[info] Benchmark had encountered error, and fail on error was requested
[error] Exception in thread "main" org.openjdk.jmh.runner.RunnerException: Benchmark caught the exception
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.Runner.runBenchmarks(Runner.java:575)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.Runner.internalRun(Runner.java:310)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.Runner.run(Runner.java:209)
[error]         at org.enso.benchmarks.common/org.enso.interpreter.bench.BenchmarksRunner.run(BenchmarksRunner.java:68)
[error]         at org.enso.runtime/org.enso.interpreter.bench.benchmarks.RuntimeBenchmarksRunner.main(RuntimeBenchmarksRunner.java:8)
[error] Caused by: org.openjdk.jmh.runner.BenchmarkException: Benchmark error during the run
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.BenchmarkHandler.runIteration(BenchmarkHandler.java:424)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.BaseRunner.runBenchmark(BaseRunner.java:261)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.BaseRunner.runBenchmark(BaseRunner.java:233)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.BaseRunner.doSingle(BaseRunner.java:138)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.BaseRunner.runBenchmarksForked(BaseRunner.java:75)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.ForkedRunner.run(ForkedRunner.java:72)
[error]         at jmh.core@1.36/org.openjdk.jmh.runner.ForkedMain.main(ForkedMain.java:86)
[error]         Suppressed: java.lang.IllegalStateException: Good!
[error]                 at org.enso.runtime/org.enso.interpreter.bench.benchmarks.semantic.ArrayProxyBenchmarks.sumOverComputingProxy(ArrayProxyBenchmarks.java:100)
[error]                 at org.enso.runtime/org.enso.interpreter.bench.benchmarks.semantic.jmh_generated.ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.sumOverComputingProxy_avgt_jmhStub(ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.java:232)
[error]                 at org.enso.runtime/org.enso.interpreter.bench.benchmarks.semantic.jmh_generated.ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.sumOverComputingProxy_AverageTime(ArrayProxyBenchmarks_sumOverComputingProxy_jmhTest.java:173)
[error]                 at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[error]                 at java.base/java.lang.reflect.Method.invoke(Method.java:565)
[error]                 at jmh.core@1.36/org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:475)
[error]                 at jmh.core@1.36/org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:458)
[error]                 at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]                 at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
[error]                 at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]                 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
[error]                 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
[error]                 at java.base/java.lang.Thread.run(Thread.java:1447)
[error] Nonzero exit code returned from runner: 1
[error] (runtime-benchmarks / Compile / run) Nonzero exit code returned from runner: 1
[error] Total time: 197 s (03:17), completed 4. 11. 2025 15:10:27
$ echo $?
1
```

--- a/test/Benchmarks/src/Table/Internal/Multi_Value_Key.enso
+++ b/test/Benchmarks/src/Table/Internal/Multi_Value_Key.enso
@@ -74,7 +74,7 @@ collect_benches = Bench.build builder->
             key_columns = table.columns
             directions = Vector.fill key_columns.length False
             make_key row_ix = Ordered_Multi_Value_Key.from_row key_columns directions row_ix
-            compare_keys key1 key2 = key1 < key2
+            compare_keys key1 key2 = Comparable.< self=key1 that=key2
             compare_ordered_keys make_key table compare_keys
 
         group_builder.specify "Primitive_Enso" <|

